tst/com/amazon/corretto/crypto/provider/test/EvpSignatureSpecificTest.java

```diff
@@ -468,10 +468,26 @@ void testRsaPSSTryUpdateParamDuringBuffer() throws Exception {
         // Finally, assert that we can set params after sign/verify reinit
         signer.setParameter(spec2);
         verifier.setParameter(spec2);
-        assertPssParamsEqual(spec2, signer.getParameters().getParameterSpec(PSSParameterSpec.class));
-        assertPssParamsEqual(spec2, verifier.getParameters().getParameterSpec(PSSParameterSpec.class));
+        assertPssParamsEqual(spec2, getPssParams(signer));
+        assertPssParamsEqual(spec2, getPssParams(verifier));
     }
 
+    // RSASSA-PSS not available on Java10, so skip the test if we can't get get a AlgorithmParameters
+    // object for it
+    private static PSSParameterSpec getPssParams(Signature signature) {
+        try {
+            return signature.getParameters().getParameterSpec(PSSParameterSpec.class);
+        } catch (Throwable e) {
+            // Unfortunately, we can't be too specific in our catch statement here
+            // because ACCP appears to sometimes sneakily throw a (checked)
```

Comment:
This sounds like a bug in ACCP that we should fix.

Reply:
yeah, it seems a bit fishy. I'm guessing that somewhere along the call stack we're not properly catching a NoSuchAlgorithmException where we should be wrapping and re-throwing with [InvalidParameterSpecException](https://docs.oracle.com/javase/7/docs/api/java/security/spec/InvalidParameterSpecException.html) ([which `getParameterSpec` throws](https://docs.oracle.com/javase/7/docs/api/java/security/AlgorithmParameters.html#getParameterSpec(java.lang.Class))).

what's really throwing me for a loop is how a checked exception (i.e. NoSuchAlgorithmException)can be thrown by a method that doesn't declare it... NoSuchAlgorithmException and InvalidParameterSpecException both share an ancestor in GeneralSecurityException, but `getParameterSpec` explicitly `throws InvalidParameterSpecException`. i've never seen this before...

I've [cut a SIM to track](https://sim.amazon.com/issues/CryptoAlg-1174), but I don't think we should gate this PR on that issue.